connect/src/com/telenor/connect/headerenrichment/HeFlowDecider.java

```diff
@@ -0,0 +1,138 @@
+package com.telenor.connect.headerenrichment;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.net.Uri;
+import android.os.Build;
+
+import java.util.Set;
+
+public class HeFlowDecider {
+
+    private static final String PARAMETER_PROMPT = "prompt";
+    private static final String PARAMETER_NO_SEAM = "no_seam";
+    private static final String PARAMETER_HE_TOKEN = "telenordigital_he_token";
+
+    /**
+     * This method is used to specify the header enrichment flow and prevent
+     * fraud possibilities. The logic that method follows is:
+     *
+     * - If network is mobile only: do header enrichment in SDK, if it fails - do
+     * header enrichment on the backend side.
+     *
+     * - If network is wifi only: block header enrichment in SDK and backend.
+     *
+     * - If network is wifi and mobile at same time: do header enrichment in SDK, if it fails -
+     * block header enrichment on the backend side.
+     *
+     * The way to ask backend services to skip header enrichment is to add "no_seam"
+     * parameter to the request.
```

Comment:
Add a comment about this code potentially overwriting prompt values provided earlier on.